src/pymodaq_plugins_thorlabs/daq_viewer_plugins/plugins_1D/daq_1Dviewer_CCSXXX.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+from pymodaq.utils.daq_utils import ThreadCommand
+from pymodaq.utils.data import DataFromPlugins, Axis, DataToExport
+from pymodaq.control_modules.viewer_utility_classes import DAQ_Viewer_base, comon_parameters, main
+from pymodaq.utils.parameter import Parameter
+from pymodaq_plugins_thorlabs.hardware.ccsxxx import CCSXXX
+
+class DAQ_1DViewer_CCSXXX(DAQ_Viewer_base):
+    """ Instrument plugin class for a 1D viewer.
+
+    This object inherits all functionalities to communicate with PyMoDAQ’s DAQ_Viewer module through inheritance via
+    DAQ_Viewer_base. It makes a bridge between the DAQ_Viewer module and the Python wrapper of a particular instrument.
+
+    Attributes:
+    -----------
+    controller: object
+        The particular object that allow the communication with the hardware, in general a python wrapper around the
+         hardware library.
+
+    """
+    params = comon_parameters + [
+        {'title': 'Integration time', 'name': 'integration_time', 'type': 'float', 'value': 100.0e-3}, # in seconds
+        {'title': 'Resource name', 'name': 'resource_name', 'type': 'str', 'value': 'USB0::0x1313::0x8087::M00934802::RAW'},
+    ]
+
+    def ini_attributes(self):
+        """Initialize attributes for the DAQ_1DViewer_CCSXXX class."""
+        self.controller: CCSXXX = None
+        self.x_axis = None
+
+    def commit_settings(self, param: Parameter):
+        """Apply the consequences of a change of value in the detector settings
+
+        Parameters
+        ----------
+        param: Parameter
+            A given parameter (within detector_settings) whose value has been changed by the user
+        """
+        if param.name() == "integration_time":
+            self.controller.set_integration_time(self.settings['integration_time'])
+
+    def ini_detector(self, controller=None):
+        """Detector communication initialization
+
+        Parameters
+        ----------
+        controller: (object)
+            custom object of a PyMoDAQ plugin (Slave case). None if only one actuator/detector by controller
+            (Master case)
+
+        Returns
+        -------
+        info: str
+        initialized: bool
+            False if initialization failed otherwise True
+        """
+        self.ini_detector_init(slave_controller=controller)
+
+        if self.is_master:
+            self.controller = CCSXXX(self.settings['resource_name'])
+            self.controller.connect()
+
+        data_x_axis = self.controller.get_wavelength_data()
+        self.x_axis = Axis(data=data_x_axis, label='Wavelength', units='nm', index=0)
+
+        self.dte_signal_temp.emit(DataToExport(name='CCSXXX',
+                                               data=[DataFromPlugins(name='Spectrum',
+                                                                     data=np.zeros(len(data_x_axis)),
```

Comment:
I'm seeing an error here I missed before: the data attribute should be a list of numpy array, so:

```
DataFromPlugins(name='Spectrum', data=[np.zeros(len(data_x_axis)),],
```